libbeat/processors/add_formatted_index/add_formatted_index.go

```diff
@@ -23,25 +23,48 @@ import (
 	"github.com/elastic/beats/v7/libbeat/beat"
 	"github.com/elastic/beats/v7/libbeat/beat/events"
 	"github.com/elastic/beats/v7/libbeat/common/fmtstr"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+func init() {
+	processors.RegisterPlugin("add_formatted_index", NewC)
+}
+
 // AddFormattedIndex is a Processor to set an event's "raw_index" metadata field
 // with a given TimestampFormatString. The elasticsearch output interprets
 // that field as specifying the (raw string) index the event should be sent to;
 // in other outputs it is just included in the metadata.
 type AddFormattedIndex struct {
 	formatString *fmtstr.TimestampFormatString
+	configString *fmtstr.TimestampFormatString
```

Comment:
Why do we need a new variable for the case where the format string comes from configuration?

Reply:
```go
	if p.configString != nil {
		index, err = p.configString.RunEvent(event)
	} else {
		index, err = p.formatString.Run(event.Timestamp)
	}
```

It seems like whether we apply the process using the whole event or just the timestamp is implicitly decided based on how we constructed the processor.

It would be more obvious if we just added an explicit flag to control the behaviour rather than inferring it like this.

Reply:
I can change it to a flag. But that is why.

Reply:
Changed to use same variable and boolean flag instead.